### PR TITLE
Make robot marker all nice and pretty

### PR DESCRIPTION
**Pull request by MortFred**

### CODE REVIEW

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        return
+    }
+    var pixelPosition = calculateObjectPixelPosition(mission.map!, currentRobotPose.position)
+    var rad = calculateNavigatorAngle(currentRobotPose, previousRobotPose)
+    drawRobotMarker(pixelPosition[0], pixelPosition[1], map)
+    drawNavigator(pixelPosition[0], pixelPosition[1], map, rad)
+}
+
+function calculateObjectPixelPosition(missionMap: MissionMap, objectPosition: ObjectPosition) {
+    var e1 = objectPosition.x
```
Comment (oysand) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:39]: Should we use let for these variables?

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+export function PlaceRobotInMap(
+    mission: Mission,
+    map: HTMLCanvasElement,
+    currentRobotPose: Pose,
+    previousRobotPose: Pose
+) {
+    if (mission.robot.pose === undefined) {
+        return
+    }
+    var pixelPosition = calculateObjectPixelPosition(mission.map!, currentRobotPose.position)
```
Comment (oysand) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:32]: Should we use let for these variables?

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    let rad = Math.atan2(
+        currentRobotPose.position.y - previousRobotPose.position.y,
+        currentRobotPose.position.x - previousRobotPose.position.x
+    )
+    rad = -rad + Math.PI / 2
+    return rad
+}
+
+function drawTagMarker(p1: number, p2: number, map: HTMLCanvasElement, tagNumber: number) {
+    var circleSize = 30
```
Comment (oysand) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:60]: Should we use let for these variables?

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+export function PlaceRobotInMap(
+    mission: Mission,
+    map: HTMLCanvasElement,
+    currentRobotPose: Pose,
+    previousRobotPose: Pose
+) {
+    if (mission.robot.pose === undefined) {
+        return
+    }
+    let pixelPosition = calculateObjectPixelPosition(mission.map!, currentRobotPose.position)
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:32]:
```suggestion
    const pixelPosition = calculateObjectPixelPosition(mission.map!, currentRobotPose.position)
```

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    let rad = calculateNavigatorAngle(currentRobotPose, previousRobotPose)
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:33]:
```suggestion
    const rad = calculateNavigatorAngle(currentRobotPose, previousRobotPose)
```

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    drawRobotMarker(pixelPosition[0], pixelPosition[1], map)
+    drawNavigator(pixelPosition[0], pixelPosition[1], map, rad)
+}
+
+function calculateObjectPixelPosition(missionMap: MissionMap, objectPosition: ObjectPosition) {
+    let e1 = objectPosition.x
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:39]: All: const

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    let c2 = missionMap.transformationMatrices.c2
+    let d1 = missionMap.transformationMatrices.d1
+    let d2 = missionMap.transformationMatrices.d2
+    let p1 = c1 * e1 + d1
+    let p2 = c2 * e2 + d2
+    return new Array(p1, p2)
+}
+
+function calculateNavigatorAngle(currentRobotPose: Pose, previousRobotPose: Pose) {
+    let rad = Math.atan2(
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:51]:
```suggestion
    const rad = Math.atan2(
```

**Review by eivindsjovold — CHANGES_REQUESTED** [review]
As a general comment i think most "let" can be "const" and most functions could be arrowfunctions

File: frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        currentRobotPose.position.y - previousRobotPose.position.y,
+        currentRobotPose.position.x - previousRobotPose.position.x
+    )
+    rad = -rad + Math.PI / 2
+    return rad
+}
+
+const drawTagMarker = (p1: number, p2: number, map: HTMLCanvasElement, tagNumber: number) => {
+    const circleSize = 30
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/MissionPage/MapPosition/MapMarkers.tsx:60]: Configurable somewhere maybe?

**Review by eivindsjovold — APPROVED** [review]
LGTM

**Pull request merged**